audit-connector/src/main/java/com/sequenceiq/cloudbreak/audit/converter/RestEventDataExtractor.java

```diff
@@ -0,0 +1,80 @@
+package com.sequenceiq.cloudbreak.audit.converter;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.audit.converter.auditeventname.rest.RestResourceAuditEventConverter;
+import com.sequenceiq.cloudbreak.audit.model.ApiRequestData;
+import com.sequenceiq.cloudbreak.audit.model.AuditEventName;
+import com.sequenceiq.cloudbreak.audit.model.EventData;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.structuredevent.event.StructuredEvent;
+import com.sequenceiq.cloudbreak.structuredevent.event.StructuredRestCallEvent;
+import com.sequenceiq.cloudbreak.structuredevent.event.rest.RestRequestDetails;
+
+@Component
+public class RestEventDataExtractor implements EventDataExtractor<StructuredRestCallEvent> {
+
+    @Value("${info.app.version:}")
+    private String cbVersion;
+
+    @Inject
+    private Map<String, RestResourceAuditEventConverter> resourceAuditEventConverters;
+
+    @Override
+    public EventData eventData(StructuredRestCallEvent structuredEvent) {
+        RestRequestDetails restRequest = structuredEvent.getRestCall().getRestRequest();
+        boolean mutating = Set.of("POST", "PUT", "DELETE").contains(restRequest.getMethod());
+        String userAgent = restRequest.getHeaders().get("user-agent");
+        return ApiRequestData.builder()
+                .withApiVersion(cbVersion)
+                .withMutating(mutating)
+                .withRequestParameters(restRequest.getRequestUri())
+                .withUserAgent(userAgent)
+                .build();
+    }
+
+    @Override
+    public AuditEventName eventName(StructuredRestCallEvent structuredEvent) {
+        String resourceType = structuredEvent.getOperation().getResourceType();
+        RestResourceAuditEventConverter restResourceAuditEventConverter = getConverter(resourceType);
+        if (restResourceAuditEventConverter != null) {
+            AuditEventName eventName = restResourceAuditEventConverter.auditEventName(structuredEvent);
+            if (eventName != null) {
+                return eventName;
+            }
+        }
+        String method = structuredEvent.getRestCall().getRestRequest().getMethod();
+        throw new UnsupportedOperationException(String.format("The `%s` with `%s` does not support for auditing", resourceType, method));
+    }
+
+    @Override
+    public Crn.Service eventSource(StructuredRestCallEvent structuredEvent) {
+        return Crn.Service.DATAHUB;
```

Comment:
how will you decide if it is a datalake?

Reply:
you can determine from the payload, the stack type either DATALAKE or WORKLOAD